pkg/catalogsourceconfig/registry.go

```diff
@@ -104,15 +106,51 @@ func (r *registry) ensureDeployment(operatorSources string) error {
 			return err
 		}
 		r.log.Infof("Created Deployment %s with registry command: %s", deployment.GetName(), registryCommand)
-	} else {
-		// Update the pod specification
+		return nil
+	}
+
+	// If the command has changed, we update the pod specification. If not, we kill the pod to
+	// force a re-sync with Quay. This is required so that we get updates from Quay when
+	// packages have not been added or removed from the spec.
+	if hasCommandChanged(deployment.Spec.Template.Spec.Containers[0].Command, registryCommand) {
 		deployment.Spec.Template = r.newPodTemplateSpec(registryCommand)
-		err = r.client.Update(context.TODO(), deployment)
+		err := r.client.Update(context.TODO(), deployment)
 		if err != nil {
 			r.log.Errorf("Failed to update Deployment %s : %v", deployment.GetName(), err)
 			return err
 		}
 		r.log.Infof("Updated Deployment %s with registry command: %s", deployment.GetName(), registryCommand)
+	} else {
+		options := &client.ListOptions{}
+		options.SetLabelSelector(r.getRawLabel())
+		pods := &core.PodList{}
+		err := r.client.List(context.TODO(), options, pods)
+		if err != nil {
+			r.log.Errorf("Failed to list pods in the Deployment %s: %v", deployment.GetName(), err)
+			return err
+		}
+
+		// There should only be one pod in the Deployment. If we find more
+		// than one we will delete the pod whose name starts with the
+		// CatalogSourceConfig's name
+		if len(pods.Items) > 1 {
+			r.log.Warnf("Found %d in Deployment %s. Will delete most likely match!",
+				len(pods.Items), deployment.GetName())
+		}
+
+		for _, pod := range pods.Items {
+			if !strings.HasPrefix(pod.GetName(), r.csc.GetName()) {
+				r.log.Warnf("Found %s in Deployment %s that did not match", pod.GetName(), deployment.GetName())
+				continue
+			}
+			if err = r.client.Delete(context.TODO(), &pod); err != nil {
+				msg := fmt.Sprintf("Failed to delete Pod %s: %v", pod.GetName(), err)
+				r.log.Errorf(msg)
+				r.log.Errorf("%s will not be updated", r.csc.GetName())
```

Comment:
Can we combine these into one error message?

Reply:
Sure, it would be become very verbose and hard to read though.